src/Voron/Data/CompactTrees/PersistentDictionary.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+using Sparrow;
+using Sparrow.Server.Compression;
+using Voron.Exceptions;
+using Voron.Global;
+using Voron.Impl;
+using static Sparrow.Hashing;
+
+namespace Voron.Data.CompactTrees
+{
+    [StructLayout(LayoutKind.Explicit, Pack = 1)]
+    public unsafe struct PersistentDictionaryHeader
+    {
+        public const int SizeOf = 32;
+
+        [FieldOffset(0)]        
+        public ulong TableHash;
+        [FieldOffset(8)]
+        public long CurrentId;
+        [FieldOffset(16)]
+        public long PreviousId;
```

Comment:
what is that for?

Reply:
This is the chain of dictionaries. In case of corruption we could still get the dictionaries through the chain and force our way to recover a dead page.

Reply:
This is currently only for indexes, so recovery is simply an issue of re-indexing.
For documents compression, we store this twice (in Voron and outside of it), to allow recovery in this manner.
But we don't have a scenario where we run recovery on indexes.